client/my-sites/hosting/staging-site-card/index.js

```diff
@@ -0,0 +1,121 @@
+import { getSiteSlug } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+const noticeOptions = {
+	duration: 3000,
+};
+
+const stagingSiteAddFailureNoticeId = 'staging-site-add-failure';
+
+const StagingSiteCard = ( { disabled, siteId, translate } ) => {
+	const { __ } = useI18n();
+	const dispatch = useDispatch();
+
+	const { data: stagingSites, isLoading: isLoadingStagingSites } = useStagingSite( siteId, {
+		enabled: ! disabled,
+	} );
+
+	const stagingSite = stagingSites && stagingSites.length ? stagingSites[ 0 ] : [];
+	const stagingSiteSlug = useSelector( ( state ) => getSiteSlug( state, stagingSite.id ) );
+
+	const showAddStagingSite = ! isLoadingStagingSites && stagingSites && stagingSites.length === 0;
+	const showManageStagingSite = ! isLoadingStagingSites && stagingSites && stagingSites.length > 0;
+
+	const transferStatus = useCheckStagingSiteStatus( stagingSite.id );
+	const isStagingSiteTransferComplete = transferStatus === transferStates.COMPLETE;
+
+	const { addStagingSite, isLoading: addingStagingSite } = useAddStagingSiteMutation( siteId, {
+		onMutate: () => {
+			dispatch( removeNotice( stagingSiteAddFailureNoticeId ) );
+		},
+		onSuccess: () => {
+			dispatch( recordTracksEvent( 'calypso_hosting_configuration_staging_site_add_success' ) );
+			dispatch( successNotice( __( 'Staging site was added.' ), noticeOptions ) );
+		},
+		onError: ( error ) => {
+			dispatch(
+				recordTracksEvent( 'calypso_hosting_configuration_staging_site_add_failure', {
+					code: error.code,
+				} )
+			);
+			dispatch(
+				errorNotice(
+					// translators: "reason" is why adding the staging site failed.
+					sprintf( __( 'Failed to add staging site: %(reason)s' ), { reason: error.message } ),
+					{
+						...noticeOptions,
+						id: stagingSiteAddFailureNoticeId,
+					}
+				)
+			);
+		},
+	} );
+
+	const getNewStagingSiteContent = () => {
+		return (
+			<div>
+				<p>
+					{ translate(
+						'A staging site is a test version of your website you can use to preview and troubleshoot changes before bringing them to your production site.'
+					) }
+				</p>
+				<Button primary disabled={ disabled || addingStagingSite } onClick={ addStagingSite }>
+					<span>{ translate( 'Create staging site' ) }</span>
+				</Button>
+			</div>
+		);
+	};
+
+	const getManageStagingSiteContent = () => {
+		return (
+			<div>
+				<p>
+					{ translate( 'Your staging site is available at {{a}}%(stagingSiteName)s{{/a}}.', {
+						args: {
+							stagingSiteName: stagingSite.url,
+						},
+						components: {
+							a: <a href={ stagingSite.url } />,
+						},
+					} ) }
+				</p>
+				<Button primary href={ `/home/${ stagingSiteSlug }` } disabled={ disabled }>
+					<span>{ translate( 'Manage staging site' ) }</span>
+				</Button>
+			</div>
+		);
+	};
+
+	return (
+		<Card className="staging-site-card">
+			<MaterialIcon icon="build" size={ 24 } />
```

Comment:
This icon is conflicting with `Web server settings`

<img width="673" alt="Screenshot 2023-02-23 at 21 27 26" src="https://user-images.githubusercontent.com/779993/221034150-a6ba0685-7c27-4389-a779-2028ff633c3c.png">

Reply:
Good catch! I tried to re-use an icon we already have in the codebase and missed that we use this one on this page. Would you have any thoughts on a good fit?

<img width="94" alt="Screenshot 2023-02-24 at 10 33 52" src="https://user-images.githubusercontent.com/727413/221143872-1273e005-39aa-4ae6-a61d-97dbcdbf35d5.png"> <img width="105" alt="Screenshot 2023-02-24 at 10 33 39" src="https://user-images.githubusercontent.com/727413/221143874-fa6ea3af-9938-4a42-87d5-55e1faff28a6.png"> <img width="111" alt="Screenshot 2023-02-24 at 10 33 33" src="https://user-images.githubusercontent.com/727413/221143876-414e6cf1-11a5-4e5b-9dc5-83169c104ee3.png"> <img width="99" alt="Screenshot 2023-02-24 at 10 33 25" src="https://user-images.githubusercontent.com/727413/221143879-e5661a39-74f6-43d7-9bab-7b1e8c538fff.png">

Reply:
CC @javierarce

Reply:
We could also use Gutenberg icons: https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library

Some suggestions:
<img width="346" alt="Screenshot 2023-02-27 at 11 06 18" src="https://user-images.githubusercontent.com/779993/221547816-9ec45fea-1dec-40a9-a609-59a3a885347e.png">

<img width="333" alt="Screenshot 2023-02-27 at 11 07 04" src="https://user-images.githubusercontent.com/779993/221547979-36b1ced5-5a64-4d22-945b-5bc11838aa73.png">
<img width="338" alt="Screenshot 2023-02-27 at 11 09 04" src="https://user-images.githubusercontent.com/779993/221548454-5e019edc-9972-47f7-9957-8ef912cae0c9.png">

<img width="345" alt="Screenshot 2023-02-27 at 11 10 58" src="https://user-images.githubusercontent.com/779993/221548828-a4a77464-6ba8-49e9-a3c5-105c0fccf41d.png">


<img width="362" alt="Screenshot 2023-02-27 at 11 08 32" src="https://user-images.githubusercontent.com/779993/221548324-f3e0b89d-d982-4738-941b-e4de4b8ab8cf.png">

Reply:
Going with the Share icon for now:

<img width="774" alt="image" src="https://user-images.githubusercontent.com/36432/221605475-707da5d9-30b9-4e69-9f14-bf70969a5e2a.png">

I created a follow up issue to explore further https://github.com/Automattic/dotcom-forge/issues/1856